tests/integration/test_integ.py

```diff
@@ -0,0 +1,129 @@
+
+
+        # Check if mdh core is running
+        is_core_up = subprocess.Popen(["docker", "container", "inspect", "-f",
+	                              "'{{.State.Running}}'", self.MDH_CORE], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, err = is_core_up.communicate()
+        if "true" not in out.decode():
+            print("MDH CORE NOT RUNNING!!!")
+            exit()
+
+        docker_compose_up=subprocess.Popen(["docker-compose", "up"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        docker_up_out, docker_up_err = docker_compose_up.communicate()
+        if docker_up_err.decode().__contains__("error") == True:
+            print("ERROR RUNNING DOCKER-COMPOSE UP!")
+            exit()
+
+        # Check if subprocess is running
+        is_up=docker_compose_up.poll()
+
+        # Returns 0 if subprocess is alive, otherwise 1
+        if is_up is not None:
+            docker_start = subprocess.Popen(
+                ["docker", "container", "start", "synthetic-file-system"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            is_up = docker_start.poll()
+            if is_up is not None:
+                print("Container synthetic-file-system cannot be started!!")
+                exit()
+        else:
+            # Added sleep because docker takes time to be set up
+            time.sleep(30)
+
+    
+    @pytest.mark.dependency()
+    def test_mount(self):
+        print("/************************************* Entered test_mount *************************************/")
+
+        # Start docker just in case it is down
+        docker_start=subprocess.Popen(
+            ["docker", "container", "start", "synthetic-file-system"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+
+        # Run mount.sh
+        docker_mount=subprocess.Popen(["docker", "container", "exec", "-d", "synthetic-file-system",
+                                         "./mount.sh"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        docker_mount_out, docker_mount_err=docker_mount.communicate()
+        docker_mount_err = docker_mount_err.decode()
+        if "failed" in docker_mount_err:
+             print("ERROR RUNNING MOUNT.SH!!!")
+             exit()
+
+        time.sleep(5)
+
+        # goes into the container and gets a list of all files under fuse_mount and writes it to a file
+        self.mounted_files=os.popen(
+            'docker container exec synthetic-file-system /bin/sh -c "cd ..; cd /fuse_mount; find . -type f" ').read()
+        f=open(self.FUSE_MOUNTED_FILES_LIST, 'w')
+        print(self.mounted_files, file=f)
+        f.close()
+
+        # Throws error if file containing mounted data is empty
+        if os.stat(self.FUSE_MOUNTED_FILES_LIST).st_size == 0:
+            print("Mounted files list at " + self.FUSE_MOUNTED_FILES_LIST, " empty!!!")
+            exit()
+
+
+
+    @pytest.mark.dependency(depends=['test_mount'])
+    def test_dir_structure(self):
```

Comment:
that is not really an integration test

Reply:
So it's not required? It checks if the files have been mounted correctly.

Reply:
that is implicitly done as you traverse the directory. more interesting is if a user can read/open files etc.

Reply:
done